### ISSUE TITLE
Seedlet: First item in Social Icons does not align with the rest in editor

### ISSUE THREAD

**Issue by iamtakashi** [Automattic/themes#2727]
## Steps to replicate

* Add Social Icons block

## Result

![Screenshot 2020-10-28 at 16 54 09](https://user-images.githubusercontent.com/908665/97469356-3acb4980-193e-11eb-84c4-f9acbdcba6a5.jpg)

## Expected

It should align with the rest. This is what we see with Varia themes.

![Screenshot 2020-10-28 at 16 55 36](https://user-images.githubusercontent.com/908665/97469540-6f3f0580-193e-11eb-9966-c25723d2277b.jpg)

**Comment by stacimc** [comment]
This is also happening for buttons:

<img width="439" alt="Screen Shot 2020-11-05 at 12 36 24 PM" src="https://user-images.githubusercontent.com/63313398/98299340-6a94d580-1f6c-11eb-9656-4689249d3880.png">

and in both cases is caused by this theme CSS (seen in the inspector):

```
.editor-styles-wrapper [data-block] [data-block]:first-child {
    margin-top: 0;
}
```

**Comment by jacklenox** [comment]
I can confirm this is also happening in Spearhead, as we'd probably expect (although in my case it's not as bad). The extent to which it looks bad is dependent on how large the value of this CSS property is set: `var(--global--spacing-vertical)`

![Screenshot from 2021-01-12 19-02-33](https://user-images.githubusercontent.com/751476/104361215-1547e800-550a-11eb-9438-70e3433a8080.png)

This is the offending code. If I comment this out, this misalignment is fixed:

https://github.com/Automattic/themes/blob/8c1e871e51e40a29de1487d7a136c4f69dd2271b/seedlet/assets/sass/blocks/utilities/_editor.scss#L153-L155

But why was this added in the first place? :thinking:

**Comment by jffng** [comment]
> But why was this added in the first place? 🤔

I believe this was added so that inserting nested blocks would not appear with extra top margin.

It may be that this is no longer necessary after some work with the lighter dom, so if it can be removed that would be great. 

More context:

> This was also an issue in Twenty Twenty-One WordPress/twentytwentyone#716. In that case it has to do with how margin is added to child blocks. It was fixed with this PR WordPress/twentytwentyone#726.

**Comment by jacklenox** [comment]
Ah, thanks Jeff! I'll investigate. :slightly_smiling_face:
